subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/OutputScrapingExecutionResult.java

```diff
@@ -220,10 +224,39 @@ public ExecutionResult assertNotOutput(String expectedOutput) {
         return this;
     }
 
+    private static class LineWithDistance {
+        private final String line;
+        private final int distance;
+
+        public LineWithDistance(String line, int distance) {
+            this.line = line;
+            this.distance = distance;
+        }
+
+        public String getLine() {
+            return line;
+        }
+
+        public int getDistance() {
+            return distance;
+        }
+    }
+
     @Override
     public ExecutionResult assertContentContains(String actualText, String expectedOutput, String label) {
         String expectedText = LogContent.of(expectedOutput).withNormalizedEol();
         if (!actualText.contains(expectedText)) {
+            if (!expectedText.contains("\n")) {
```

Comment:
Do we check somewhere that the expected text is not multiline - that wouldn't ever find a match in the actual due to the line splitting.

Reply:
I don't understand your question. This is where we check if expected text is single-line for exactly the reason you mention.

Reply:
My mistake, sorry.  I was looking for an `else` case and missed that the `failOnMissingOutput` still runs in this situation.